### PR TITLE
Set up initial repository structure

### LINKED ISSUE #2: Decide on repository structure
We need to decide on how to structure the repository.

As an initial suggestion possibly something like 

```
├── README.md
├── cpp
│   └── README.md
├── fortran
│   └── README.md
├── java
│   └── README.md
├── julia
│   └── README.md
├── matlab
│   └── README.md
├── python
│   └── README.md
└── r
    └── README.md
```

with per-language README.md giving human readable descriptions of recommended standards and/or linking to the external standards we recommend? We can then also possibly add a per language linter configuration to each directory where relevant.

With this sort of structure we could then have site both directly viewable on GitHub and use Jekyll GitHub pages integration to also allow viewing at https://github-pages.arc.ucl.ac.uk/coding-standards/ possibly using something like https://github.com/benbalter/jekyll-readme-index to simplify URLs

Edit: Updated `c++` to `cpp` to avoid having to encode `+` in URLs

### PR DESCRIPTION
**Pull request by matt-graham**
Resolves #2 

Assumes we will have a top-level directory per language, each with an associated `README.md` file describing or linking to the conventions we recommend. As discussed in https://github.com/UCL-ARC/coding-standards/issues/2#issuecomment-1770910877 we can also put other language specific content such as recommended linter configuration files in the relevant directories and link to from the README files.

Currently this has empty `README.md` files for an arbitrary set of languages that I first thought of, other than Python which copies over the existing material from internal documentation. We might want to just remove the empty files and add directories as we add each language, but I'll hold off making changes here till @milanmlft's R recommendations from #8 are merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,11 @@
...
+
+UCL Advanced Research Computing Centre's recommended coding standards for research software projects
+
+  * [C++](cpp)
+  * [Fortran](fortran)
+  * [Java](java)
+  * [Julia](julia)
+  * [MATLAB](matlab)
+  * [Python](python)
+  * [R](r)
```
Comment (samcunliffe) [inline README.md:11]:
```suggestion
  * [Python](python)
  * [R](r)
```

I might hide everything at the start and unhide as the `lang/README.md`'s become populated.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,11 @@
-# coding-standards
-ARC's recommended coding standards
+# Coding standards
+
+UCL Advanced Research Computing Centre's recommended coding standards for research software projects
```
Comment (samcunliffe) [inline README.md:3]:
```suggestion
UCL Advanced Research Computing Centre's recommended coding standards for research software projects.
These are our default assumed style choices unless there are existing project- or domain-specific local style conventions.
```

Comment (samcunliffe) [reply README.md:3]: I guess because we're not _forcing_ this on anyone (or are we?)

File: python/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Python
+
+Python code will adhere to the [PEP8](http://legacy.python.org/dev/peps/pep-0008/) Python coding standard.
```
Comment (samcunliffe) [inline python/README.md:3]:
```suggestion
Python code should, as a minimum, adhere to the [PEP8](http://legacy.python.org/dev/peps/pep-0008/) Python coding standard.
We also encourage compliance with modern style recommendations, such as type annotations in [PEP484](https://peps.python.org/pep-0484/).
```
...?
@paddyroddy ?

**Review by samcunliffe — APPROVED** [review]
I know it's only a draft but this all LGTM.

A question about where we want to be on the rule-recommendation spectrum. We probably want to iterate a bit on this in a separate discussion/ticket/pull request.

File: python/README.md (hunk 3 above)
Comment (matt-graham) [reply python/README.md:3]: I'd be happy to add PEP484 / type annotations as a recommendation, previous material was just a verbatim copy from what was already in internal docs. I'll add this suggestion for now with the view that we can always iterate!

File: README.md (hunk 1 above)
Comment (matt-graham) [reply README.md:11]: Agreed better to have only links to READMEs that are actually populated. I'll also remove the placeholder files, mainly added these to test how things displayed on GitHub pages.

File: python/README.md (hunk 3 above)
Comment (paddyroddy) [reply python/README.md]: Sounds good

**Pull request merged**